### PR TITLE
feat(@theme-ui/prism): Support multiple highlight wrappers in a single code block

### PR DESCRIPTION
**Pull request by Yihwan**
Allows multiple `// highlight-start` and `// highlight-end` wrappers in a single code block. Previously, attempting to add multiple wrappers would cause subsequent highlights to be a few lines off: 

```js
  // highlight-start
  console.log('hello, world!') 
  // highlight-end
  let other = "no highlight"
  // highlight-start
  console.log('hello, world!')
  // highlight-end
  other = "still no highlight here"
  // oh no
```

Screen Shot 2020-12-27 at 4 54 53 PM

I also edited the docs to be more explicit on how to use this feature. It was hard to tell what was going on from the Rich Text view because the actual highlight comments were being removed. 

I noticed that the `.highlight` class names seemed to be applying correctly, but not any highlight-specific styles. I'm not sure if that is expected, but I left that alone for now. 🙂

Screen Shot 2020-12-27 at 6 47 18 PM

### CODE REVIEW

File: CHANGELOG.md
Comment (lachlanjc) [inline CHANGELOG.md]:
```suggestion
- `@theme-ui/prism`: Support multiple highlight wrappers in a single code block. PR #1393
```

**Review by lachlanjc — APPROVED** [review]
Thanks so much for the PR! I’ll leave it to @hasparus for a more thorough review but everything here looks great.

File: packages/prism/src/index.tsx
Comment (hasparus) [inline packages/prism/src/index.tsx]:
```suggestion
             * so we can keep track of multiple highlight-start and highlight-end blocks. 
```

**Comment by hasparus** [comment]
Hi @Yihwan! Thanks for the PR. It looks great. Comments make sense, and I love the addition to the docs.

**Review by hasparus — APPROVED** [review]

**Comment by Yihwan** [comment]
@hasparus @lachlanjc woohoo! feel free to ping me if there's any need for follow-up work on this or anything tangential.

**Comment by lachlanjc** [comment]
@hasparus Can we merge? I'm not the code owner here so deferring to you.

**Pull request merged**
